Form/Type/DocumentType.php

```diff
@@ -34,6 +57,68 @@ public function getLoader(ObjectManager $manager, $queryBuilder, $class)
         );
     }
 
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $choiceListFactory = $this->choiceListFactory;
+        $choiceLoaders = &$this->choiceLoaders;
+        $type = $this;
+
+        $choiceLoader = function (Options $options) use ($choiceListFactory, &$choiceLoaders, $type) {
```

Comment:
can you add a comment on the lines where we differ from the orm loader to explain the differences? will make future maintenance easier.